llvm/test/Transforms/InstCombine/select.ll

```diff
@@ -1,5 +1,5 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
-; RUN: opt < %s -passes=instcombine -S | FileCheck %s
+; RUN: opt < %s -passes='instcombine<no-verify-fixpoint>' -S | FileCheck %s
```

Comment:
If I add `inbounds` flag for `ptr getelementptr (i8, ptr @arr, i64 4)`, `%cmp` will be constant folded into `false` before visiting the select.